### PR TITLE
 Implement parser for rational literal types

### PR DESCRIPTION
**Pull request by JorikSchellekens**

### CODE REVIEW

File: src/types/ast/rational_literal.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Range } from "../../misc";
+import { TypeNode } from "./type";
+
+interface Rational {
+  numerator: bigint;
+  denominator: bigint;
+}
+
+export class RationalLiteralType extends TypeNode {
+    public readonly literal?: Rational;
```
Comment (cd1m0) [inline src/types/ast/rational_literal.ts:10]: Question: Why is the literal member optional? Looking at the changes in  the grammar in typeString_grammar.pegjs literal is always provided.

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,3159 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```
Comment (cd1m0) [inline yarn.lock:2]: Would you mind removing this file from the commit?

**Review by cd1m0 — COMMENTED** [review]
Thank you for your PR! Good work! Just 2 small nits in comments, and one small request. Could you add the following line:

```
     1.1;
```

After this line:

https://github.com/ConsenSys/solc-typed-ast/blob/master/test/samples/solidity/types/types.sol#L48

The idea is that if we have a rational literal in that file, then the following unit test will exercise the new code you added:

https://github.com/ConsenSys/solc-typed-ast/blob/master/test/unit/types/typestrings.spec.ts

**Review by blitz-1306 — CHANGES_REQUESTED** [review]
Thanks for the contribution. I'm mostly +1 with the above comments from @cd1m0. If there would be any blockers for you to introduce the changes, just inform us. I will consider picking up the branch and doing the rest in that case.

File: src/types/ast/rational_literal.ts (hunk 1 above)
Comment (JorikSchellekens) [reply src/types/ast/rational_literal.ts:10]: I had this same question about `IntLiteralType` and presumed there was some usecase I wasn't aware of.

**Comment by JorikSchellekens** [comment]
> Thanks for the contribution. I'm mostly +1 with the above comments from @cd1m0. If there would be any blockers for you to introduce the changes, just inform us. I will consider picking up the branch and doing the rest in that case.

Thanks, should be good to go?

**Review by cd1m0 — APPROVED** [review]
lgtm

File: yarn.lock (hunk 2 above)
Comment (blitz-1306) [reply yarn.lock]: Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/98/commits/19f312848b466885b4cdcb3fff2eb5dd7ec6c813

File: src/types/ast/rational_literal.ts (hunk 1 above)
Comment (blitz-1306) [reply src/types/ast/rational_literal.ts]: Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/98/commits/19f312848b466885b4cdcb3fff2eb5dd7ec6c813

**Review by blitz-1306 — APPROVED** [review]
Good to go. Thanks again for your efforts.

**Pull request merged**